pinot-core/src/main/java/org/apache/pinot/core/plan/FilterPlanNode.java

```diff
@@ -109,8 +109,17 @@ private BaseFilterOperator constructPhysicalOperator(FilterContext filter,
           DataSource dataSource = _indexSegment.getDataSource(lhs.getIdentifier());
           switch (predicate.getType()) {
             case TEXT_MATCH:
-              return new TextMatchFilterOperator(dataSource.getTextIndex(),
-                  ((TextMatchPredicate) predicate).getValue(), _numDocs);
+              if (dataSource.getTextIndex() != null) {
+                  return new TextMatchFilterOperator(
+                          dataSource.getTextIndex(),
+                          ((TextMatchPredicate) predicate).getValue(),
+                          _numDocs);
+              } else {
+                  return new TextMatchFilterOperator(
```

Comment:
If I understand correctly, the FST can only be used to solve the regex matching query. If that is the case, we should use it for `REGEXP_LIKE` instead of `TEXT_MATCH`

Reply:
+1